#The differences included in this patch are taken from bug Ticket 58790
#(https://bugreports.qt.io/browse/QTBUG-58790), these changes are included
#in the current 5.8 branch (but not in the 5.8.0 release).  At such a time
#that a patched version of 5.8 is released (or once these changes make their
#way into the 5.9 codebase) this patch can be removed.
--- a/qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/input.py	2017-01-03 09:28:53.000000000 +0000
+++ b/qtwebengine/src/3rdparty/chromium/tools/gyp/pylib/gyp/input.py	2017-05-15 20:22:06.553594247 +0000
@@ -898,12 +898,23 @@
         else:
           # Fix up command with platform specific workarounds.
           contents = FixupPlatformCommand(contents)
+
+          osenv = os.environ
+
+          if (osenv.has_key('PKG_CONFIG_SYSROOT_DIR')
+               and osenv['PKG_CONFIG_SYSROOT_DIR'] == ''):
+             osenv = dict(os.environ)
+             del osenv['PKG_CONFIG_SYSROOT_DIR']
+          else:
+             osenv = None
+
           try:
             p = subprocess.Popen(contents, shell=use_shell,
                                  stdout=subprocess.PIPE,
                                  stderr=subprocess.PIPE,
                                  stdin=subprocess.PIPE,
-                                 cwd=build_file_dir)
+                                 cwd=build_file_dir,
+                                 env=osenv)
           except Exception, e:
             raise GypError("%s while executing command '%s' in %s" %
                            (e, contents, build_file))
--- a/qtwebengine/src/3rdparty/chromium/media/capture/video/linux/v4l2_capture_delegate.cc	2017-01-03 09:28:53.000000000 +0000
+++ b/qtwebengine/src/3rdparty/chromium/media/capture/video/linux/v4l2_capture_delegate.cc	2017-05-15 20:22:10.531471465 +0000
@@ -10,6 +10,10 @@
 #include <sys/mman.h>
 #include <utility>
 
+#if !defined(OS_OPENBSD)
+#include <linux/version.h>
+#endif
+
 #include "base/bind.h"
 #include "base/files/file_enumerator.h"
 #include "base/posix/eintr_wrapper.h"
@@ -252,9 +256,12 @@
 
   // Set anti-banding/anti-flicker to 50/60Hz. May fail due to not supported
   // operation (|errno| == EINVAL in this case) or plain failure.
-  if ((power_line_frequency_ == V4L2_CID_POWER_LINE_FREQUENCY_50HZ) ||
-      (power_line_frequency_ == V4L2_CID_POWER_LINE_FREQUENCY_60HZ) ||
-      (power_line_frequency_ == V4L2_CID_POWER_LINE_FREQUENCY_AUTO)) {
+  if ((power_line_frequency_ == V4L2_CID_POWER_LINE_FREQUENCY_50HZ)
+      || (power_line_frequency_ == V4L2_CID_POWER_LINE_FREQUENCY_60HZ)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,0)
+      || (power_line_frequency_ == V4L2_CID_POWER_LINE_FREQUENCY_AUTO)
+#endif
+    ) {
     struct v4l2_control control = {};
     control.id = V4L2_CID_POWER_LINE_FREQUENCY;
     control.value = power_line_frequency_;
--- a/qtwebengine/src/3rdparty/chromium/media/capture/video/linux/video_capture_device_linux.cc	2017-01-03 09:28:53.000000000 +0000
+++ b/qtwebengine/src/3rdparty/chromium/media/capture/video/linux/video_capture_device_linux.cc	2017-05-15 20:22:10.532471633 +0000
@@ -17,6 +17,7 @@
 #include <sys/videoio.h>
 #else
 #include <linux/videodev2.h>
+#include <linux/version.h>
 #endif
 
 namespace media {
@@ -142,8 +143,12 @@
     case media::PowerLineFrequency::FREQUENCY_60HZ:
       return V4L2_CID_POWER_LINE_FREQUENCY_60HZ;
     default:
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,2,0)
       // If we have no idea of the frequency, at least try and set it to AUTO.
       return V4L2_CID_POWER_LINE_FREQUENCY_AUTO;
+#else
+      return V4L2_CID_POWER_LINE_FREQUENCY_60HZ;
+#endif
   }
 }
 
